modules/nlu/src/backend/api.ts

```diff
@@ -48,8 +41,10 @@ export default async (bp: typeof sdk, state: NLUState) => {
   router.post('/cross-validation/:lang', async (req, res) => {
     const { botId, lang } = req.params
     const ghost = bp.ghost.forBot(botId)
+    const entityService = new EntityService(ghost, botId)
```

Comment:
Instead of creating a new instance of the service (and passing it the ghost) everytime, what do you think of initializing it once and passing it to nluByBot? there you could access the bot's engine, entityService and a future intentService

Reply:
we could yes, I ended up with a class because it was annoying to pass the ghost in every function param. But then I realized creating the instance everywhere was quite a hustle as well. I can add this in botState